--- a/packstack/plugins/prescript_000.py
+++ b/packstack/plugins/prescript_000.py
@@ -133,6 +133,17 @@ def initConfig(controllerObject):
                    "USE_DEFAULT"     : False,
                    "NEED_CONFIRM"    : False,
                    "CONDITION"       : False },
+                  {"CMD_OPTION"      : "exclude-servers",
+                   "USAGE"           : "Comma separated list of servers to be excluded from installation in case you are running Packstack the second time with the same answer file and don't want Packstack to touch these servers. Leave plain if you don't need to exclude any server.",
+                   "PROMPT"          : "Enter a comma separated list of server(s) to be excluded. Leave plain if you don't need to exclude any server.",
+                   "OPTION_LIST"     : [],
+                   "DEFAULT_VALUE"   : '',
+                   "MASK_INPUT"      : False,
+                   "LOOSE_VALIDATION": False,
+                   "CONF_NAME"       : "EXCLUDE_SERVERS",
+                   "USE_DEFAULT"     : False,
+                   "NEED_CONFIRM"    : False,
+                   "CONDITION"       : False },
                  ]
     groupDict = { "GROUP_NAME"            : "GLOBAL",
                   "DESCRIPTION"           : "Global Options",
